CRM/ManualDirectDebit/Hook/PostProcess/Contribution/DirectDebitMandate.php

```diff
@@ -84,10 +123,43 @@ private function redirectToContributionTab() {
     $this->form->controller->setDestination(CRM_Utils_System::url('civicrm/contact/view', http_build_query([
         'action' => 'browse',
         'reset' => 1,
-        'cid' => $this->form->getVar('_entityId'),
+        'cid' => $this->currentContactId,
         'selectedChild' => 'contribute',
       ])
     ));
   }
 
+  /**
+   * Sends mail if appropriate checkbox is checked on
+   */
+  public function checkMailNotification() {
+    if (!isset($this->form->getVar('_submitValues')['send_mandate_update_notification_to_the_contact']) || empty($this->form->getVar('_submitValues')['send_mandate_update_notification_to_the_contact'])) {
+      return FALSE;
+    }
+
+    $valueOfSendMailCheckbox = $this->form->getVar('_submitValues')['send_mandate_update_notification_to_the_contact'];
+    $isSendMailCheckboxTurnedOn = $valueOfSendMailCheckbox == 1;
+
+    if ($isSendMailCheckboxTurnedOn) {
+
+//      TODO: Sending mail process will be implemented in DD-30
```

Comment:
We really shouldn't merge into develop things that are half done. To be able to finish this, I think we should base this branch (DD-31) off the DD-30 branch, and treat DD-30 as a feature branch or epic. If we do this, the merge of this branch would go into DD-30 branch, and not develop, and we'd merge DD-30 into develop once everything is stable.

Exmple:
```
|
* Merge DD-30 to develop
|\
| * Merge DD-31 to DD-30
| |\
| | |
| | * DD-31
| | | 
| |/
| * DD-30
| | 
|/
| 
| Develop
```

CC @guanhuan